frontend/src/lib/k8s/cluster.ts

```diff
@@ -17,13 +17,29 @@ import StatefulSet from './statefulSet';
 
 export const HEADLAMP_ALLOWED_NAMESPACES = 'headlamp.allowed-namespaces';
 
-function getAllowedNamespaces() {
-  const cluster = getCluster();
-  if (!cluster) {
+/**
+ * Gives an optionally configured list of allowed namespaces.
+ *
+ * @param cluster Optional cluster to check for allowed namespaces.
+ *                If not given the current cluster allowed name spaces are used.
+ *
+ * @returns A list of configured name spaces for the given cluster or current cluster.
+ *          If a zero length list, then no allowed namespace has been configured for cluster.
+ *          If length > 0, allowed namespaces have been configured for this cluster.
+ *
+ * There are usecases where a user doesn't have the authority to list
+ * all the namespaces. In that case it becomes difficult to access things
+ * around Headlamp. To prevent this we can allow the user to pass a set
+ * of namespaces they know they have access to and we can use this set to
+ * make requests to the API server.
+ */
+export function getAllowedNamespaces(cluster?: string | undefined): string[] {
```

Comment:
You can assign a default value right in the definition, then you won't need `theCluster` variable
 `getAllowedNamespaces(cluster: string | undefined = getCluster())`

Reply:
Good idea, updated. Thanks.